src/download.c

```diff
@@ -156,18 +166,44 @@ void *swupd_curl_parallel_download_start(size_t max_xfer, size_t max_xfer_bottom
 
 	h->max_xfer = max_xfer;
 	h->max_xfer_bottom = max_xfer_bottom;
-	h->success_cb = success_cb;
 	h->curl_hashmap = hashmap_new(SWUPD_CURL_HASH_BUCKETS, file_hash_cmp, file_hash_value);
 	h->timeout = RETRY_TIMEOUT;
 
-	fprintf(stderr, "Starting download of remaining update content. This may take a while...\n");
-
 	return h;
 error:
 	free(h);
 	return NULL;
 }
 
+/*
+ * Set parallel downloads callbacks.
+ *
+ *  - success_cb(): Called for each sucessfull download, where data is the data
+ *                  informed on swupd_curl_parallel_download_enqueue().
+ *                  success_cb should return true if the download file was
+ *                  handled correctly. Return false to schedule a retry.
+ *  - error_cb():   Called for each failed download, where data is the data
+ *                  informed on swupd_curl_parallel_download_enqueue()
+ *                  and response is the HTTP response code. error_cb should
```

Comment:
I think you are missing something here: 

>  error_cb should ?? if the error was handled by caller or false to schedule a retry.

Reply:
Probably I removed that by accident. 
error_cb should return true...